File: pract9_Hotel_Service_Java/pract9/HotelClient.java
import java.rmi.registry.LocateRegistry;
import java.rmi.registry.Registry;
import java.util.Scanner;

public class HotelClient {

    public static void main(String[] args) {

        try {

            // Connect to registry
            Registry registry = LocateRegistry.getRegistry("localhost", 6001);

            // Lookup remote object
            HotelServiceInterface hotel =
                    (HotelServiceInterface) registry.lookup("HotelService");

            Scanner sc = new Scanner(System.in);

            int choice = 0;

            while (choice != 4) {

                System.out.println("\n===== HOTEL BOOKING SYSTEM =====");
                System.out.println("1. Book a Room");
                System.out.println("2. Cancel Booking");
                System.out.println("3. View Bookings");
                System.out.println("4. Exit");

                System.out.print("Enter your choice: ");
                choice = sc.nextInt();
                sc.nextLine();

                switch (choice) {

                    case 1:

                        System.out.print("Enter Guest Name: ");
                        String guestBook = sc.nextLine();

                        System.out.println(
                                hotel.bookRoom(guestBook)
                        );

                        break;

                    case 2:

                        System.out.print("Enter Guest Name: ");
                        String guestCancel = sc.nextLine();

                        System.out.println(
                                hotel.cancelBooking(guestCancel)
                        );

                        break;

                    case 3:

                        System.out.println(
                                hotel.viewBookings()
                        );

                        break;

                    case 4:

                        System.out.println("Exiting...");
                        break;

                    default:

                        System.out.println("Invalid choice!");
                }
            }

            sc.close();

        } catch (Exception e) {

            System.out.println("Client Error: " + e.getMessage());
        }
    }
}

File: Pract9_Hotel_Service/hotel_client.py
import Pyro4

def main():

    ns = Pyro4.locateNS()

    uri = ns.lookup("hotel.booking")

    hotel = Pyro4.Proxy(uri)

    while True:

        print("\n===== HOTEL BOOKING SYSTEM =====")
        print("1. Book Room")
        print("2. Cancel Booking")
        print("3. View Bookings")
        print("4. Exit")

        choice = input("Enter your choice: ")

        if choice == "1":

            guest = input("Enter guest name: ")

            result = hotel.book_room(guest)

            print(result)

        elif choice == "2":

            guest = input("Enter guest name: ")

            result = hotel.cancel_booking(guest)

            print(result)

        elif choice == "3":

            bookings = hotel.view_bookings()

            print("\nCurrent Bookings:")

            if bookings:
                for guest, status in bookings.items():
                    print(f"{guest} --> {status}")
            else:
                print("No bookings available.")

        elif choice == "4":

            print("Exiting Client...")
            break

        else:
            print("Invalid Choice")


if __name__ == "__main__":
    main()